src/main/java/uk/gov/hmcts/ccd/v2/external/controller/DocumentController.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.ccd.v2.external.controller;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.ccd.domain.model.definition.Document;
+import uk.gov.hmcts.ccd.domain.service.common.UIDService;
+import uk.gov.hmcts.ccd.domain.service.stdapi.DocumentsOperation;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadRequestException;
+import uk.gov.hmcts.ccd.v2.V2;
+import uk.gov.hmcts.ccd.v2.external.resource.DocumentsResource;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/cases")
+public class DocumentController {
+
+    private final UIDService caseReferenceService;
+    private final DocumentsOperation documentsOperation;
+
+    @Autowired
+    public DocumentController(
+        final UIDService caseReferenceService,
+        final DocumentsOperation documentsOperation
+    ) {
+        this.documentsOperation = documentsOperation;
+        this.caseReferenceService = caseReferenceService;
+    }
+
+    @GetMapping(
+        path = "/{caseId}/documents",
+        headers = {
+            V2.EXPERIMENTAL_HEADER
+        },
+        produces = {
+            V2.MediaType.CASE_DOCUMENTS
+        }
+    )
+    @ApiOperation(
+        value = "Retrieve case documents by case ID",
+        notes = V2.EXPERIMENTAL_WARNING
+    )
+    @ApiResponses({
+        @ApiResponse(
+            code = 200,
```

Comment:
It could be good to provide response details from a Catalog, where response structures and wordings could be standardised. But that would be outside the intent of this PR.